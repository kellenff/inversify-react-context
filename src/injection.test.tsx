import React from 'react';
import 'reflect-metadata';
import { Container, interfaces } from 'inversify';
import renderer from 'react-test-renderer';
import { ContainerProvider, useInjection } from './index';

describe('injection hook', () => {
  let container: Container,
      identifier: interfaces.ServiceIdentifier<symbol>,
      Component: React.FC;

  beforeEach(() => {
    container  = new Container();
    identifier = Symbol('name');

    container.bind<string>(identifier).toConstantValue('world');
    Component = () => {
      const value = useInjection<string>(identifier);

      return (
        <p>{ value }</p>
      );
    };
  });

  it('should return the value if it is bound', () => {
    const rendered = renderer.create(
      <ContainerProvider container={ container }>
        <Component />
      </ContainerProvider>,
          ),
          tree     = rendered.toJSON();

    expect(tree).toMatchSnapshot();
    // @ts-ignore
    expect(tree?.children).toEqual(['world']);
  });

  it('should throw if the value if unbound', () => {
    const OtherComponent: React.FC = () => {
            const nonValue = useInjection<string>(Symbol('nonValue'));

            return (
              <p>{ nonValue }</p>
            );
          };

    const render = () => {
      const rendered = renderer.create(
        <ContainerProvider container={ container }>
          <OtherComponent />
        </ContainerProvider>,
      );

      rendered.toJSON();
    };

    expect(render).toThrow(' matching bindings found for serviceIdentifier: Symbol(');
  });
});
