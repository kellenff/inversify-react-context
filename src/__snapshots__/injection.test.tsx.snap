// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injection hook should return the value if it is bound 1`] = `
<p>
  world
</p>
`;
